crates/task-impls/src/transactions.rs

```diff
@@ -361,16 +349,51 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, V: Versions> TransactionTask
 
         let (_, precompute_data) = precompute_vid_commitment(&[], membership_total_nodes);
 
-        // Broadcast the empty block
+        Some(PackedBundle::new(
+            vec![].into(),
+            metadata,
+            block_view,
+            vec1::vec1![null_fee],
+            Some(precompute_data),
+            Some(TYPES::AuctionResult::default()),
+        ))
+    }
+
+    #[allow(clippy::too_many_lines)]
+    /// marketplace view change handler
+    pub async fn handle_view_change_marketplace(
+        &mut self,
+        event_stream: &Sender<Arc<HotShotEvent<TYPES>>>,
+        block_view: TYPES::Time,
+    ) -> Option<HotShotTaskCompleted> {
+        let version = match self.upgrade_lock.version(block_view).await {
+            Ok(v) => v,
+            Err(err) => {
+                error!("Upgrade certificate requires unsupported version, refusing to request blocks: {}", err);
+                return None;
+            }
+        };
+
+        let (parent_view, parent_hash) = self.last_vid_commitment(block_view).await?;
```

Comment:
Can we add a retry loop on this similar to what we have in legacy? It's not uncommon for the leader not to have the relevant VID right away.
Also, if we fail to find it, we should still produce an empty block.

Reply:
Updated: I created a wrapper `last_vid_commitment_retry` for the loop, and used that in both places.